.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,66 @@
+
+name: LibLCM CI and CD
+
+on:
+  pull_request:
+    branches: [ develop ]
+  push:
+    branches: [ develop ]
+
+jobs:
+
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      DOTNET_NOLOGO: true
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.4
+      with:
+        fetch-depth: 0
+
+    # Install the .NET Core workload
+    - name: Install .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 6.0.x
+
+    - name: Install mono-devel package
+      run: sudo apt-get install mono-devel
+      if: matrix.os == 'ubuntu-latest'
+
+    - name: Install icu-fw package
+      run: wget -qO - http://linux.lsdev.sil.org/downloads/sil-testing.gpg | sudo tee /etc/apt/trusted.gpg.d/linux-lsdev-sil-org.asc &&
+        sudo add-apt-repository "deb http://linux.lsdev.sil.org/ubuntu $(lsb_release -sc)-experimental main" &&
+        sudo apt-get install icu-fw
+      if: matrix.os == 'ubuntu-latest'
+
+    # Solve problems with CustomIcuFallbackTests: the current windows-latest image (2022) includes this file that breaks our tests.
+    - name: Solve problems
+      run: Remove-Item c:\tools\php\icuuc70.dll -Force
+      if: matrix.os == 'windows-latest'
+
+    - name: Build
+      run: dotnet build
+
+    - name: Test
+      run: dotnet test --no-restore --no-build -p:ParallelizeAssembly=false
+
+    - name: Package
+      run: dotnet pack --include-symbols --no-restore --no-build -p:SymbolPackageFormat=snupkg 
+
+    - name: Publish to Nuget
+      run: dotnet nuget push artifacts/*.nupkg -s https://api.nuget.org/v3/index.json -k ${{secrets.SILLSDEV_PUBLISH_NUGET_ORG}} --skip-duplicate
+      if: github.event_name == 'push' && matrix.os == 'ubuntu-latest'
```

Comment:
Do we have to check the branch name as well so that we don't publish to nuget on a push to a feature branch?